docs/agented-vm-guide.md

```diff
@@ -0,0 +1,267 @@
+    Ips:
+      Address:       10.0.1.45
+      Address Type:  InternalIP
+      Address:       54.193.202.222
+      Address Type:  ExternalIP
+    Mac:             02:96:fc:89:17:ed
+    Name:            eni-056b0d6da3f592943
+  Provider:          AWS
+  State:             running
+  Tags:
+    Name:                 ubuntu2004
+  Virtual Private Cloud:  vpc-0d6bb6a4a880bd9ad
+Events:                   <none>
+```
+
+## ExternalNode CR Creation
+
+The `Nephe Controller` will create an ExternalNode CR corresponding to each
+VirtualMachine CR, which has the `agented` flag enabled. In the below example,
+since VM `i-019459b33d951b62e` is configured with agented flag set to true,
+a corresponding ExternalNode `virtualmachine-i-019459b33d951b62e` is created.
+
+```bash
+kubectl get en -A
+```
+
+```text
+# Output
+NAMESPACE   NAME                                 AGE
+vm-ns       virtualmachine-i-019459b33d951b62e   6s
+```
+
+## ExternalEntities CR Creation
+
+For each ExternalNode CR, `antrea-controller` will create a corresponding
+ExternalEntity CR. In this example, the ExternalEntity
+`virtualmachine-i-019459b33d951b62e-5df40` corresponds to the ExternalNode
+`virtualmachine-i-019459b33d951b62e`, while the other three ExternalEntities
+are created by `nephe-controller` with ExternalNode field set to
+`nephe-controller`.
+
+```bash
+kubectl get ee -A
+```
+
+```text
+# Output
+NAMESPACE   NAME                                       AGE
+vm-ns       virtualmachine-i-0033eb4a6c846451d         9s
+vm-ns       virtualmachine-i-019459b33d951b62e-5df40   9s
+vm-ns       virtualmachine-i-05e3fb66922d56e0a         9s
+vm-ns       virtualmachine-i-0a20bae92ddcdb60b         9s
+```
+
+## Install Antrea-Agent On Public Cloud VM
+
+Nephe provides a wrapper install script, to facilitate easier installation on
+the public cloud VMs. The wrapper install script sets the Environment variable
+`NODE_NAME` to be same as ExternalNode name. It downloads the `antrea-agent`
+install script from antrea repository and triggers installation.
+
+The wrapper install script requires 4 arguments:
+Namespace            - Specifies the Namespace to be used by the `antrea-agent`.
+It should match with the ExternalNode Namespace.
+Antrea version       - Specifies the Antrea version to be used.
+Kubeconfig           - Provides access to the Kubernetes API server.
+Antrea Kubeconfig    - Provides access to the Antrea API server.
+
+For more information on how to generate the kubeconfig files, please refer to
+antrea [ExternalNode](https://github.com/antrea-io/antrea/blob/main/docs/external-node.md#install-antrea-agent-on-vm)
+documentation.
+
+### Installation On Linux VMs
+
+Download the [wrapper install script](../hack/install-wrapper.sh) and run
+installation.
+
+```bash
+./install-wrapper.sh --ns vm-ns --antrea-version v1.8.0 \
+  --kubeconfig /root/antrea-agent.kubeconfig \
+  --antrea-kubeconfig /root/antrea-agent.antrea.kubeconfig --bin /root/antrea-agent
```

Comment:
may be remove this from documentation ?

Reply:
I just wanted to show an example of how to pass various arguments to the script.
I can remove it if its not required.

Reply:
I changed it to use relative path and update the PR.
`
./install-wrapper.sh --ns vm-ns --antrea-version v1.8.0 --kubeconfig ./antrea-agent.kubeconfig \
  --antrea-kubeconfig ./antrea-agent.antrea.kubeconfig --bin ./antrea-agent
`

`
.\install-wrapper.ps1 -Namespace vm-ns -AntreaVersion v1.8.0 -KubeConfigPath .\antrea-agent.kubeconfig `
  -AntreaKubeConfigPath .\antrea-agent.antrea.kubeconfig
`